### ISSUE TITLE
Cannot specify ContainerEnvironmentVariable via the command line using -p:ContainerEnvironmentVariable=?

### ISSUE THREAD

**Issue by lymberl** [dotnet/sdk-container-builds#451]
For automating builds and deploys it would be useful if we can specify container variables via the command line. Such as:
`dotnet publish --os linux --arch x64 -p:ContainerEnvironmentVariable=EnvironmentVariable=Value /t:PublishContainer -c Release`

How can this be done via the command line since that property is a complex property? 
`<ContainerEnvironmentVariable Include="ASPNETCORE_ENVIRONMENT" Value="Development" />`

But when building for other environments i would like to set it to a different value such as the below:
`<ContainerEnvironmentVariable Include="ASPNETCORE_ENVIRONMENT" Value="Staging" />`

Are there any workarounds or other means to achieve this?

**Comment by baronfel** [comment]
Hi @lymberl - currently this is not possible. MSBuild doesn't allow passing Items on the command line. @rainersigwald are there patterns that we could rely on here? Or I suppose in the worst case we could do something like

* have a property that users could set
* have a target that explicitly 'splits' that property into Items by parsing a specific format.

**Comment by rainersigwald** [comment]
> * have a property that users could set
> * have a target that explicitly 'splits' that property into Items by parsing a specific format.

This is the best I can think of too; it's nice and explicit.

**Comment by baronfel** [comment]
How confusing would it be if we chose the Property name of `ContainerEnvironmentVariable` as well? Does MSBuild handle the ambiguity correctly based on how the name is used, either in `@()` for items or `$()` for Properties?

Otherwise I suppose that would impact the choice of name for the property to use here.

**Comment by rainersigwald** [comment]
Yes, we have totally separate lookup tables so you can have a property and an item with the same name. It tends to be pretty confusing to debug (even more than getting it wrong when there's only one but it's the other kind) but in this case I think that's outweighed by the consistency of naming.

**Comment by almostjulian** [comment]
We need something like this for CI/CD, as is we need to stick to docker based container builds to inject environment variables.

**Comment by SViradiya-MarutiTech** [comment]
Is it resolved?, we are facing the same issue on providing `ContainerEnvironmentVariable` in the command line to set `ASPNETCORE_ENVIRONMENT` Environment variable.

**Comment by acraven** [comment]
@lymberl I may have a workaround for you which should work in CI/CD...

Add this to your services` csproj files if you haven't already.

```csproj
  <ItemGroup>
    <ContainerEnvironmentVariable Include="MINIMUMDBVERSION" Value="$(MinDbVersion)" />
  </ItemGroup>
```

Set a env var in whatever console and publish the service

```ps1
$env:MinDbVersion=123

dotnet publish --os linux --arch x64 /t:PublishContainer -c Release

docker inspect apiservice
```

```json
    "Env": [
        "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
        "APP_UID=1654",
        "ASPNETCORE_HTTP_PORTS=8080",
        "DOTNET_RUNNING_IN_CONTAINER=true",
        "DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=true",
        "DOTNET_VERSION=9.0.6",
        "ASPNET_VERSION=9.0.6",
        "MINIMUMDBVERSION=123"
    ],
```

For anyone else using `aspire publish` preview for Docker Compose (Aspire.Hosting.Docker 9.3.1-preview.1.25305.6) swap out `dotnet publish` for `aspire publish`

**Comment by m3ac-AllbrittenJ** [comment]
bumping to note that others are still stumbling into this

**Comment by SMayer135** [comment]
What is the expected way to set ASPNETCORE_ENVIRONMENT if through the command line isn't supported?
